react-components/src/components/AccountContainer/__test__/accountContainer.test.js

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { MockedProvider } from '@apollo/react-testing';
+import { I18nextProvider } from 'react-i18next';
+import { render } from '@testing-library/react';
+
+import UserContextProvider from '../../../context/UserContext';
+import { CartProvider } from '../../Minicart/cartContext';
+import i18n from '../../../../__mocks__/i18nForTests';
+
+import AccountContainer from '../accountContainer';
+
+describe('<AccountContainer>', () => {
+    beforeAll(() => {
```

Comment:
Please review the way you use mocking here. Make sure that the mocked method is restored after the test. See for example https://jestjs.io/docs/en/mock-functions#mocking-modules.

Reply:
Hi @herzog31, are we saying the mockClear() here like:
```
    afterEach(() => {
        ReactDOM.createPortal.mockClear();
    });
```
?

Reply:
Please check https://jestjs.io/docs/en/mock-function-api#mockfnmockclear. `mockClear` doesn't restore the original functionality. It resets the recorded calls. I'm referring to restore the original functionality :) 

Try
```
jest.mock('react-dom');
ReactDOM.createPortal.mockImplementation(e => e);
```

Reply:
Hi @herzog31, I see what you mean, I tried with jest.mock('react-dom'), but then I receive the error reported by test script, it says as below:
![image](https://user-images.githubusercontent.com/5010963/87001790-5ed01700-c1f7-11ea-9d09-80cff638e550.png)
So I can't mock the whole `react-dom` module. I tried to use `spyOn`, which works quite well, so we can have:
```
    beforeAll(() => {
        // mock createPortal because we don't have the DOM element to render the AccountDropdown
        jest.spyOn(ReactDOM, 'createPortal').mockImplementation(element => element);
    });
```
Are you happy with this?
The other consideration is, I actually get the current implementation from `react-components/src/components/CartTrigger/__test__/cartTrigger.test.js`, not sure we should change the implementation there as well to keep consistent?

Reply:
That works as well 👍 
We can fix the implementation in the other test later.

Reply:
I've changed the `accountContainer.test.js` to use `jest.spyOn()` then :)